src/librustc_mir/borrow_check/mod.rs

```diff
@@ -1552,21 +1579,55 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
                 if let Err(place_err) = self.is_mutable(place, is_local_mutation_allowed) {
                     error_reported = true;
 
-                    let item_msg = match self.describe_place(place) {
-                        Some(name) => format!("immutable item `{}`", name),
-                        None => "immutable item".to_owned(),
-                    };
-
-                    let mut err = self.tcx.cannot_assign(span, &item_msg, Origin::Mir);
-                    err.span_label(span, "cannot mutate");
+                    let mut err_info = None;
+
+                    match *place_err {
+
+                        Place::Projection(box Projection {
+                        ref base, elem:ProjectionElem::Deref}) => {
+
+                            match *base {
+                                Place::Local(local) => {
+                                    let locations = self.mir.find_assignments(local);
+                                        if locations.len() > 0 {
+                                            let item_msg = if error_reported {
+                                                self.get_secondary_err_msg(base)
+                                            } else {
+                                                self.get_default_err_msg(place)
+                                            };
+                                            
```

Comment:
A line consisting of only spaces is also considered "trailing whitespaces".

Reply:
@kennytm can you help out with the fixes for the tidy errors? I've tried certain things but they don't seem to work. One of the errors is also in the previously written code.

Reply:
@gaurikholkar The line numbers seems wrong. You could run `tidy` locally by

```
./x.py test src/tools/tidy
```

The lines with trailing whitespaces are:

```
tidy error: src/librustc_mir/borrow_check/mod.rs:1532: trailing whitespace
tidy error: src/librustc_mir/borrow_check/mod.rs:1536: trailing whitespace
tidy error: src/librustc_mir/borrow_check/mod.rs:1595: trailing whitespace
```

Reply:
Thanks you @kennytm . Yeah the line nos were wrong. My local tidy script panics -> [logs](https://gist.github.com/gaurikholkar/186eb92e2f3057ee4228e9931d7a1ca6)

Reply:
@gaurikholkar you should just remove the entire `binaryen` folder, it is no longer used.